### PR TITLE
[wallet-ext] new select dropdown

### PR DESCRIPTION
**Pull request by Nikhil-Mysten**
## Description 

full dropdown

https://www.figma.com/file/dCHKynDnLAAQyoYSU6G16Y/02-Enoki-Shared?node-id=1%3A7791&mode=dev

## Test Plan 
locally

How did you test the new or updated feature?
---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/wallet/src/ui/app/components/SelectDropdown/index.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+export function SelectDropdown({
+	placeholder,
+	dropdownOptions,
+	onValueChange,
+	value,
+}: SelectDropdownProps) {
+	return (
+		<div className="z-5">
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/SelectDropdown/index.tsx:22]: Does the parent need this or can you just give this to the portal content?

File: apps/wallet/src/ui/app/components/SelectDropdown/index.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+			<Select.Root onValueChange={onValueChange} value={value}>
+				<Select.Trigger
+					className="flex items-center border border-solid border-gray-45 shadow-md rounded-md bg-transparent px-4 py-2 gap-1.5 focus:outline-none"
+					aria-label="Food"
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/SelectDropdown/index.tsx:26]: This is wrong, we should pass this through the parent.

File: apps/wallet/src/ui/app/components/SelectDropdown/index.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ChevronDown12 } from '@mysten/icons';
+import * as Select from '@radix-ui/react-select';
+import { Text } from '_app/shared/text';
+
+export interface SelectDropdownProps {
+	placeholder?: string;
+	dropdownOptions?: string[];
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/SelectDropdown/index.tsx:10]: I think `options` is probably sufficient, I don't think the `dropdown` prefix gives us anything here.

File: apps/wallet/src/ui/app/components/SelectDropdown/index.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+					<Select.Icon className="flex items-center">
+						<ChevronDown12 className="text-steel" />
+					</Select.Icon>
+				</Select.Trigger>
+				<Select.Portal>
+					<Select.Content
+						className="z-10 min-w-[100px] bg-transparent"
+						position="popper"
+						side="bottom"
+						sideOffset={-35}
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/SelectDropdown/index.tsx:42]: Curious if this number works for all possible sizes of the dropdown?

File: apps/wallet/src/ui/app/components/SelectDropdown/index.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import { Text } from '_app/shared/text';
+
+export interface SelectDropdownProps {
+	placeholder?: string;
+	dropdownOptions?: string[];
+	onValueChange?: (value: string) => void;
+	value?: string;
+}
+
+export function SelectDropdown({
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/SelectDropdown/index.tsx:15]: Can you link the figma for this? Just curious where this will be used

**Review by mamos-mysten — APPROVED** [review]

**Pull request merged**